web/packages/teleterm/src/ui/services/deepLinks/deepLinksService.ts

```diff
@@ -88,13 +88,70 @@ export class DeepLinksService {
         break;
       }
       case '/authenticate_web_device': {
-        // TODO (avatus): add authenticate device web confirmation dialog
-        // this case is a stub and will not be reached
+        const id = result.url?.searchParams?.id;
+        const token = result.url?.searchParams?.token;
+        await this.askAuthorizeDeviceTrust(result.url, id, token);
         break;
       }
     }
   }
 
+  private async askAuthorizeDeviceTrust(
+    url: DeepURL,
+    id: string,
+    token: string
+  ): Promise<void> {
+    const rootClusterId = url.hostname;
+    const clusterAddress = url.host;
+    const prefill = {
+      clusterAddress,
+      username: url.username,
+    };
+
+    const rootClusterUri = routing.getClusterUri({ rootClusterId });
+    const rootCluster = this.clustersService.findCluster(rootClusterUri);
+
+    if (!rootCluster) {
```

Comment:
This part is copied from the connect my computer handler. I think we should abstract it but I wasn't sure if we should extra the "login to root cluster AND get desired workspace" or if I should do them separately. Thoughts?

Reply:
It doesn't matter much. But I think usually you'll want to always do both actions when using a deep link, so that the user has a better view into which workspace the given deep link is related to.

Reply:
i abstracted and created `loginAndSetDesiredWorkplace` which returns the values we are using after that chain of events